app/room/[roomId]/page.js

```
@@ -0,0 +1,327 @@
+    try {
+      const offer = await webrtcUtils.createOffer(peerConnectionRef.current);
+      signalingRef.current.sendOffer(offer);
+    } catch (error) {
+      console.error('Error creating offer:', error);
+    }
+  };
+  
+  const cleanupResources = () => {
+    // Stop all media tracks
+    if (localStreamRef.current) {
+      localStreamRef.current.getTracks().forEach(track => track.stop());
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)